rim/templates/rim/edit.html

```diff
@@ -3,59 +3,97 @@
                         {{ field.label_tag }}
                         {{ field.errors }}
                         {{ field }}
                     </div>
-                {% endwith %}
-                <div class="cell large-9 cell-block-y">
-                    <div class="grid-x">
-                        {% for field in form.visible_fields|slice:"1:" %}
-                            <div class='small-4 large-4 addcell cell {% if field.field.required %}required{% endif %}'>
-                                {{ field.label_tag }}
-                                {{ field.errors }}
-                                {{ field }}
-                            </div>
-                            {% if not forloop.first and not forloop.last and forloop.counter|divisibleby:3 %}
-                                </div><div class="grid-x">
-                            {% endif %}
-                        {% endfor %}
-                    </div>
+                    {% if not forloop.first and not forloop.last and forloop.counter|divisibleby:3 %}
+                </div>
+                <div class="grid-x">
+                    {% endif %}
+                    {% endfor %}
+                </div>
+            </div>
+            {% for hidden in form.hidden_fields %}
+            {{ hidden }}
+            {% endfor %}
+
+        </div>
+        <div class='cell large-9 cell-block-y'>
+            <div class="cell cell-block-y">
+                <div class="grid-x" id="display_inputs">
+
                 </div>
             </div>
-            <div class='grid-x'>
-                <div class='small-3 large-3 cell'>
-                    <button type='submit' class='button' id="submit_button">{% trans 'Submit' %}</button>
+            <div class="cell cell-block-y">
+                <div class="grid-x" id="storage_inputs">
+
                 </div>
             </div>
-        </form>
-    </div>
-{% endblock %}
+            <div class="cell cell-block-y">
+                <div class="grid-x" id="gpu_inputs">
+
+                </div>
+            </div>
+            <div class="cell cell-block-y">
+                <div class="grid-x" id="nic_inputs">
+
+                </div>
+            </div>
+            <div class="grid-x" style="justify-content: center;">
+                <button type="button" class="button" onclick="addInputs('display')">Add Display</button>
+                <button type="button" class="button" onclick="addInputs('storage')">Add Storage</button>
+                <button type="button" class="button" onclick="addInputs('gpu')">Add GPU</button>
+                <button type="button" class="button" onclick="addInputs('nic')">Add NIC</button>
+
+            </div>
+        </div>
+
+        <div class='grid-x'>
+            <div class='small-3 large-3 cell'>
+                <button type='button' class='button' id="submit_button" onclick="submitAddEditForm()">Submit</button>
```

Comment:
Try not to remove translation tags e.g. `{% trans 'Submit' %}`. This will make things much easier when/if we go about translating this.

Also, there are a number of other static strings in this file that could benefit from translation tags.